### ISSUE TITLE
Black Screen then crash when launching with min-ed-launcher

### ISSUE THREAD

**Issue by hugeblank** [rfvgyhn/min-ed-launcher#84]
**Describe the bug/error**
After downloading/installing/updating edh4 through min-ed-launcher then attempting to launch, A black screen appears for about 5 seconds, then vanishes and closes without warning. Initially I had used the vanilla installer to download horizons, but min-ed didn't recognize it as installed, so I followed the process somewhere in one of the closed issues (I recall it being in the 50's) to force it to reinstall/update. That seemed to get min-ed to see it, but now I'm facing this issue.

Another thing I noticed while following that closed issue set of instructions. There's this directory that looks like it's new horizons that's not named `elite-dangerous-odyssey-64`:
![Screenshot_20230207_184616](https://user-images.githubusercontent.com/16393543/217415628-3265994c-cf11-4021-9508-29486d01cc3e.png)
I assume this is the directory that the vanilla installer installed into, and is the reason why min-ed didn't originally pick it up?

**Debug log**
[min-ed-launcher.log](https://github.com/rfvgyhn/min-ed-launcher/files/10681199/min-ed-launcher.log)

**Comment by hugeblank** [comment]
Ah! And to further confirm my suspicions, I swapped the two directories, renaming `elite-dangerous-odyssey-64` to `elite-dangerous-odyssey-64-old`, then renaming `FORC-FDEV-DO-38-IN-40` to `elite-dangerous-odyssey-64`. This version launched into the game correctly, but was out of date, even though the update checker ran over it.

**Comment by OzDruiD** [comment]
I too have many installed directories, being from a Lifetime backer account, then a Steam key cutover just to keep things simple. - my account LONG outdates the offical steam ccutover- so technically sstill a "frontier" account. 

This can be imcredibly annoying, as even though STEAM updates the game, or Min-Launcher updates the game files - if the Vanilla ED-Launcher is run, it still the thinks the game files are out of date and downloads again... so I get 3 downloads every update  Steam, Mini Launcher, and ED-Launcher\Vanilla.      Where is this all updating?  The ONE folder \SteamApps\Common\EliteDangerous.   

In the \products folder of that,  directories I have
elite-dangerous-65
elite-dangterous-odyssey-64
FORC-FDEV-DO-1000  
I only memtion as my FORC FDEV is a different folder name.

I run my min-launcher in a separate folder entirely and whilst you CAN specificy a gamepath in the mini launcher conflig file, mine is being detected automatically.  I only hope to help with these comments, if i find anything more I will post as well, but I know different 'versionss' of the game from Frontier and Steam matter - I will try and find the post around that in my own issue bug here on github.

**Comment by hugeblank** [comment]
> 3. Vanilla installed copy of EDH4 that is moved to where min-ed expects it is not up to date, even though min-ed launcher claims it to be.

To follow up on this: 
a) vanilla launcher detected the renamed directory, so I didn't have to rename it back.
b) vanilla launcher was able to recognize that the product was out of date and successfully downloaded and installed the required update, unlike min-ed.
c) Instead of launching through vanilla, I tried min-ed this time, and it worked, connected and all!

Of course this is far from optimal, min-ed should be doing everything I did by hand and through the vanilla launcher. Of the two things that it's meant to do for me. It, with a thorough amount of coaxing, *barely* did one.

**Comment by OzDruiD** [comment]
@hugeblank Autodection seemed to work for me, so I am curious to expand on that, is your install in a non-standard drive, folder path etc,  like it should work, however - it does provide the ability to specify non standard.   Some unrelated issue, but possible other clues are part of issue 83, https://github.com/rfvgyhn/min-ed-launcher/issues/83  - where it raised the thing for me ..Acccounts pre-dating April 2019 have a different experience as well .. although this may have no influence on your auto-detection of correct folder install issue - the pathing etc might.

**Comment by hugeblank** [comment]
Nothing too non-standard about my install besides it being on Linux. It's through Steam, bought the game at some point before April 2022. I have all of my steam games on a library folder that's on a hard drive, off my main boot ssd to save space.

**Comment by rfvgyhn** [comment]
While I haven't had a chance to look into the the actual problem, min-ed-launcher _should_ handle the folders that are named by SKU. It seems like that's broken at the moment though.

I haven't thought of a logical reason for why FDev implemented it this way, but the launcher decides the product's folder location differently if it's running via steam/epic. When using steam/epic, it checks for products/[directory] (the value of [directory] comes from the purchases api response which can be seen in your log file) and if that doesn't exist, it checks for products/[SKU].  When using a frontier account, it does the reverse. You should be able to have only one folder per product, either [directory] or [SKU] and both launchers _should_ be able to figure things out. My guess is there's a bug in min-ed-launcher in that logic.

@hugeblank As a temporary workaround, you might try deleting your [cache](https://github.com/rfvgyhn/min-ed-launcher#cache) and then checking if min-ed-launcher can properly update the game.

**Comment by hugeblank** [comment]
> @hugeblank As a temporary workaround, you might try deleting your [cache](https://github.com/rfvgyhn/min-ed-launcher#cache) and then checking if min-ed-launcher can properly update the game.

Alright, I've cleared it, the next time an update comes around we'll see if it's able to update properly.

**Comment by hugeblank** [comment]
> I haven't thought of a logical reason for why FDev implemented it this way, but the launcher decides the product's folder location differently if it's running via steam/epic. 
Yeah I was wondering why there's 2 different possible locations for the same product too. They're making things so much more difficult for themselves.

> When using a frontier account, it does the reverse. You should be able to have only one folder per product, either [directory] or [SKU] and both launchers _should_ be able to figure things out. My guess is there's a bug in min-ed-launcher in that logic.
I have a question regarding this. I got the game through steam, but I believe I still needed to create a frontier account in order to install/launch the game. Is this not normal?

**Comment by rfvgyhn** [comment]
> I have a question regarding this. I got the game through steam, but I believe I still needed to create a frontier account in order to install/launch the game. Is this not normal?

Yes, that's normal. It would have more accurate for me to say a frontier license. Before April 2019, when purchasing from Steam, you'd get both a Frontier and Steam license to play the game. You also have a Frontier license if you buy the game from their store.
